scripts/Tools/preview_namelists

```diff
@@ -15,50 +15,38 @@ from CIME.utils             import expect
 import argparse, doctest
 
 ###############################################################################
-def parse_command_line(args, description):
+def parse_command_line(args):
 ###############################################################################
-    parser = argparse.ArgumentParser(
-        usage="""\n%s [--debug]
-OR
-%s --verbose
-OR
-%s --help
-OR
-%s --test
-
-\033[1mEXAMPLES:\033[0m
-    \033[1;32m# Run \033[0m
-    > %s
-""" % ((os.path.basename(args[0]), ) * 5),
-
-description=description,
-
-formatter_class=argparse.ArgumentDefaultsHelpFormatter
-)
-
+    cime_model = CIME.utils.get_model()
+    parser = argparse.ArgumentParser()
     CIME.utils.setup_standard_logging_options(parser)
 
     parser.add_argument("caseroot", nargs="?", default=os.getcwd(),
                         help="Case directory to build")
+    parser.add_argument('--component',
+                        help="Specify component's namelist to build.")
+    parser.add_argument('--test', action='store_true',
+                        help="Run preview_namelist in test mode.")
 
-    args = parser.parse_args(args[1:])
+    args = parser.parse_args()
 
     CIME.utils.handle_standard_logging_options(args)
 
-    return args.caseroot
+    return args
 
 ###############################################################################
 def _main_func(description):
 ###############################################################################
-    if ("--test" in sys.argv):
+    args = parse_command_line(sys.argv)
```

Comment:
> you're going to have to unpack at some point, as we'd never want API that expects the args object. Why not unpack right away and be done with it?

(1) At least with the scripts I've dealt with, the particular functions called by the main program typically only take a few of the total command-line arguments. Having, say, 4 positional arguments is less error-prone than having a 20-element positional list.

(2) Even if the main program turns around and immediately passes all arguments to some other routine, this can be done via keyword arguments rather than positional arguments - so again, much less error-prone.

> Another benefit, you don't have to prefix everything with args.

I'd gladly type a few extra characters for each argument if it frees me from the stress-inducing modification of these long position-dependent lists.